_im-plugin/index-context.md

```diff
@@ -0,0 +1,170 @@
+3. Enable setting `cluster.application_templates.enabled` to `true`. See [configuring settings]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/index/#static-settings) for steps on how to do this.
+
+
+### Create Index
+
+For example, if you are using your indices to store metrics data, you can declare `metrics` as a context on your index:
+
+````
+```json
+PUT /my-metrics-index
+{
+  "context": {
+    "name": "metrics"
+  }
+}
+```
+{% include copy-curl.html %}
+````
+
+
+After creation, the index will have the context added to it, and also few settings applied:
+
+
+````
+```json
+GET /my-metrics-index
+```
+{% include copy-curl.html %}
+````
+
+````
+```json
+{
+    "my-metrics-index": {
+        "aliases": {},
+        "mappings": {},
+        "settings": {
+            "index": {
+                "codec": "zstd_no_dict",
+                "refresh_interval": "60s",
+                "number_of_shards": "1",
+                "provided_name": "my-metrics-index",
+                "merge": {
+                    "policy": "log_byte_size"
+                },
+                "context": {
+                    "created_version": "1",
+                    "current_version": "1"
+                },
+                ...
+            }
+        },
+        "context": {
+            "name": "metrics",
+            "version": "_latest"
+        }
+    }
+}
+```
+````
+
+#### Considerations
+
+There are few restrictions which are applied when you decide to use `context` param when creating your indices:
+
+1. When you create an index using `context` field, any settings which are declared in context cannot be provided as part of index creation, or dynamic settings updates.
+2. Once set on an index/index-template, context cannot be removed from it.
+
+The above points ensure that at any point, the recommended settings/mappings are always applied on the indices.
+
+
+
+### Create Index Templates
+
+You can also provide the `context` parameter while creating index templates:
+
+````
+```json
+PUT _index_template/my-logs
+{
+    "context": {
+        "name": "logs",
+        "version": "1"
+    },
+    "index_patterns": [
+        "my-logs-*"
+    ]
+}
+```
+{% include copy-curl.html %}
+````
+
+````
+```json
+GET _index_template/my-logs
+```
+{% include copy-curl.html %}
+````
+
+````
+```json
+{
+    "index_templates": [
+        {
+            "name": "my-logs2",
+            "index_template": {
+                "index_patterns": [
+                    "my-logs1-*"
+                ],
+                "context": {
+                    "name": "logs",
+                    "version": "1"
+                }
+            }
+        }
+    ]
+}
+```
+````
+
+
+This will create an index template with the context information as `logs`, and all indices created via this index template will get the metadata provided by the associated component template.
```

Comment:
check other documentation what have we used "indices" or "indexes"?

Reply:
We use "indexes". I'll make this switch when I modify the file.